libs/MobileSync/MobileSync/Classes/Util/CompositeRequestHelper.swift

```diff
@@ -0,0 +1,330 @@
+           case .UPSERT:
+               return RestClient.shared.requestForUpsert(withObjectType: objectType, externalIdField: externalIdFieldName!, externalId: externalId, fields: fields!, apiVersion: nil)
+           case .DELETE:
+               return RestClient.shared.requestForDelete(withObjectType: objectType, objectId: objectId!, apiVersion: nil)
+           }
+        }
+        
+        func  asDictForCollectionRequest() -> Dictionary<String, Any> {
+            var record:Dictionary<String, Any> = Dictionary()
+            record["attributes"] = ["type": objectType]
+            if let fields = fields {
+                for (fieldName, fieldValue) in fields {
+                    record[fieldName] = fieldValue
+                }
+            }
+            
+            if (requestType == .UPDATE) {
+                record["Id"] = objectId
+            }
+            
+            if (requestType == .UPSERT) {
+                if let externalIdFieldName = externalIdFieldName {
+                    record[externalIdFieldName] = externalId
+                }
+            }
+            
+            return record
+        }
+        
+        @objc
+        static func requestForCreate(objectType:String, fields:Dictionary<String, Any>) -> RecordRequest {
+            return RecordRequest(requestType:.CREATE, objectType: objectType, fields: fields, objectId: nil, externalId: nil, externalIdFieldName: nil)
+        }
+
+        @objc
+        static func requestForUpdate(objectType:String, objectId:String, fields:Dictionary<String, Any>) -> RecordRequest {
+            return RecordRequest(requestType:.UPDATE, objectType: objectType, fields: fields, objectId: objectId, externalId: nil, externalIdFieldName: nil)
+        }
+
+        @objc
+        static func requestForUpsert(objectType:String, externalIdFieldName:String, externalId:String, fields:Dictionary<String, Any>) -> RecordRequest {
+            return RecordRequest(requestType:.UPSERT, objectType: objectType, fields: fields, objectId: nil, externalId: externalId, externalIdFieldName: externalIdFieldName)
+        }
+
+        @objc
+        static func requestForDelete(objectType:String, objectId: String) -> RecordRequest {
+            return RecordRequest(requestType:.DELETE, objectType: objectType, fields: nil, objectId: objectId, externalId: nil, externalIdFieldName: nil)
+        }
+        
+        static func getRefIds(recordRequests:Array<RecordRequest>, requestType:RequestType) -> Array<String> {
+            return recordRequests
+                .filter { $0.requestType == requestType }
+                .map { $0.referenceId! }
+        }
+
+        static func getIds(recordRequests:Array<RecordRequest>, requestType:RequestType) -> Array<String> {
+            return recordRequests
+                .filter { $0.requestType == requestType && $0.objectId != nil }
```

Comment:
I think you could also do this if you wanted to get around the nil check/force unwrap
```
 .filter { $0.requestType == requestType }
 .compactMap { $0.objectId }
 ```